common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsAuthorizationRequest.java

```diff
@@ -22,7 +22,325 @@
+    private static final String PLATFORM_VALUE = "MSAL.Android";
+    private static final String PROMPT_LOGIN = "login";
+    private static final String PROMPT_SELECT_ACCOUNT = "select_account";
+    private static final String PROMPT_CONSENT = "consent";
+
+    private String mAuthorizationEndpoint;
+    private URL mAuthority;
+    private UUID mCorrelationId;
+    private final Set<String> mExtraScopesToConsent = new HashSet<>();
+    /**
+     * Optional. Indicates the type of user interaction that is required. The only valid values at this time are 'login', 'none', and 'consent'.
+     */
+    private MicrosoftStsPromptBehavior mPromptBehavior;
+    /**
+     * Optional. Can be used to pre-fill the username/email address field of the sign-in page for the user, if you know their username ahead of time.
+     */
+    private String mLoginHint;
+    private String mUid;
+    private String mUtid;
+    private String mDisplayableId;
+    /**
+     *  Used to secure authorization code grants via Proof Key for Code Exchange (PKCE) from a native client.
+     */
+    private PKCEChallenge mPKCEChallenge;
+    private String mExtraQueryParam;
+    private String mSliceParameters;
+    private String mLibraryVersion;
+
+    public UUID getCorrelationId() {
+        return mCorrelationId;
+    }
+
+    public void setCorrelationId(final UUID correlationId) {
+        mCorrelationId = correlationId;
+    }
+
+    public String getAuthorizationEndpoint() {
+        return mAuthorizationEndpoint;
+    }
+
+    public void setAuthorizationEndpoint(final String authorizationEndpoint) {
+        mAuthorizationEndpoint = authorizationEndpoint;
+    }
+
+    public Set<String> getExtraScopesToConsent() {
+        return mExtraScopesToConsent;
+    }
+
+    public String getLoginHint() {
+        return mLoginHint;
+    }
+
+    public void setLoginHint(final String loginHint) {
+        mLoginHint = loginHint;
+    }
+
+    public String getUid() {
+        return mUid;
+    }
+
+    public void setUid(final String uid) {
+        mUid = uid;
+    }
+
+    public String getUtid() {
+        return mUtid;
+    }
+
+    public void setUtid(final String utid) {
+        mUtid = utid;
+    }
+
+    public String getDisplayableId() {
+        return mDisplayableId;
+    }
+
+    public void setDisplayableId(final String displayableId) {
+        mDisplayableId = displayableId;
+    }
+
+    public MicrosoftStsPromptBehavior getPromptBehavior() {
+        return mPromptBehavior;
+    }
+
+    public void setPromptBehavior(final MicrosoftStsPromptBehavior promptBehavior) {
+        mPromptBehavior = promptBehavior;
+    }
+
+    public PKCEChallenge getPKCEChallenge() {
+        return mPKCEChallenge;
+    }
+
+    public void setPKCEChallenge(final PKCEChallenge pKCEChallenge) {
+        mPKCEChallenge = pKCEChallenge;
+    }
+
+    public String getExtraQueryParam() {
```

Comment:
In another spot we've called this `getExtraQP` -- pick one and pull it up to an interface and/or superclass